node/coinstacks/solana/api/src/controller.ts

```diff
@@ -126,4 +127,30 @@ export class Solana implements BaseAPI {
       throw handleError(err)
     }
   }
+
+  /**
+   * Get the current recommended gas fees to use in a transaction
+   *
+   * @returns {Promise<number>} current fees specified in lamports including base fee
+   */
+  @Response<BadRequestError>(400, 'Bad Request')
+  @Response<ValidationError>(422, 'Validation Error')
+  @Response<InternalServerError>(500, 'Internal Server Error')
+  @Post('/fees/estimate')
+  async getEstimateFees(@Body() body: GasFeesBody): Promise<number> {
+    try {
+      const deserializedMessage = Message.from(Buffer.from(body.message, 'base64'))
```

Comment:
use versioned message to support both legacy and v0 messages
```suggestion
      const deserializedMessage = VersionedMessage.deserialize(Buffer.from(body.message, 'base64'))
```